### ISSUE TITLE
webapp: Channel Editor:  incorrect path to channels directory

### ISSUE THREAD

**Issue by white-haired-uncle** [MythTV/mythtv#1005]
* **Platform**:
`Linux mythbackend 6.5.0-35-generic #35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue May  7 09:00:52 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
`
* **MythTV version**:

```
$ mythbackend --version           
Please attach all output as a file in bug reports.
MythTV Version : v35-Pre-331-g41b921c949-dirty
MythTV Branch : master
Network Protocol : 91
Library API : 35.20220913-1
QT Version : 5.15.3
Options compiled in:
 linux profile use_hidesyms using_alsa using_oss using_pulse using_pulseoutput using_backend using_bindings_perl using_bindings_python using_bindings_php using_dvb using_firewire using_frontend using_hdhomerun using_satip using_vbox using_ceton using_joystick_menu using_libcec using_libcrypto using_gnutls using_libdns_sd using_libxml2 using_lirc using_mheg using_opengl using_egl using_qtwebkit using_qtscript using_qtdbus using_taglib using_v4l2 using_v4l2prime using_x11 using_system_libbluray using_system_libudfread using_systemd_notify using_systemd_journal using_drm using_bindings_perl using_bindings_python using_bindings_php using_freetype2 using_mythtranscode using_opengl using_egl using_drm using_vaapi using_nvdec using_vdpau using_ffmpeg_threads using_mheg using_libass using_libxml2 using_libmp3lame

```

* **Package version**:

* **Component**:

### What steps will reproduce the bug?

Note "null/channels" just above the Icon Search.  (I think we used to see "null" in one of the paths to $something in the grabber config as well, I'm not sure if it was ~mythtv or ~mythtv/.mythtv, but it looks from here like pretty much the same issue -- #976 perhaps)

![Screenshot_2024-12-27_12-17-52](https://github.com/user-attachments/assets/97dd23a5-3f73-49da-9550-eb07fac21492)

### Additional information

BE running as mythtv, so I assume it's looking for ~mythtv/.mythtv/channels/

```
$ ps ux | grep mythback            
mythtv       879  0.8  3.8 3599196 73020 ?       Ssl  Oct17 844:33 /usr/local/bin/mythbackend --quiet --syslog local7
mythtv    566449  0.0  0.1   9212  2432 pts/3    S+   12:20   0:00 grep mythback
$ ls -ld ~mythtv/.mythtv/channels/     
drwxrwxr-x 2 mythtv mythtv 12288 Dec 26 03:48 /home/mythtv/.mythtv/channels/

```
I probably should try to stay more up to date if I'm going to be making reports on master.

**Comment by white-haired-uncle** [comment]
Based on the comment in #976, I checked that mythtv has $HOME set.

I also exported MYTHCONFDIR=/home/mythtv/.mythtv in ~mythtv/.kshrc (Yes, ksh, not bash) and that did not help.

Then, with MYTHCONFDIR still set, I stopped mythbackend via systemctl and started it again without that systemd nonsense.  That worked.

> Select an icon from directory "/home/mythtv/.mythtv/channels" for this channel.

Clearly the fix is to upgrade to Devuan
